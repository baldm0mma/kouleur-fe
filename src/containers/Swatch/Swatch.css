.swatch {
  flex: 1;
}

.color-swatch{
  height: 50vh;
  transition: all 2s ease;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 1rem;
}


#lock, #hex{
  opacity: .5;
  opacity: .5;
  font-size: 2rem;
  align-self: center;
} 