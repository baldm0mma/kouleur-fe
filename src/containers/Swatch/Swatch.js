import React from 'react';
import { connect } from 'react-redux';
import { setCurrentPalette, toggleLock } from '../../actions';
import './Swatch.css';
import { Icon } from 'semantic-ui-react';

const  Swatch = ({ hex, isLocked, toggleLock, num }) => {

  const chooseFontColor = hex => {
    var r = parseInt(hex.substring(0, 2), 16); 
    var g = parseInt(hex.substring(2, 4), 16); 
    var b = parseInt(hex.substring(4, 6), 16); 
    return r * 0.299 + g * 0.587 + b * 0.114 > 186 ? '#000000' : '#ffffff';
  };

  return (
    <article className='swatch'>
      <section
        className='color-swatch'
        style={{ backgroundColor: `#${hex}` }}
        onClick={() => toggleLock(num)}
      >
        {isLocked && (
          <Icon
            name='lock'
            style={{ color: `${chooseFontColor(hex)}` }}
            id='lock'
            size='big'
          />
        )}
        {!isLocked && (
          <Icon
            name='lock open'
            style={{ color: `${chooseFontColor(hex)}` }}
            id='lock'
            size='big'
          />
        )}
        <p style={{ color: `${chooseFontColor(hex)}` }} id='hex'>
          #{hex.toUpperCase()}
        </p>
      </section>
    </article>
  );
}


export const mapDispatchToProps = dispatch => ({
  setCurrentPalette: currentPalette => dispatch(setCurrentPalette(currentPalette)),
  toggleLock: num => dispatch(toggleLock(num))
});

export default connect(
  null,
  mapDispatchToProps
)(Swatch);
