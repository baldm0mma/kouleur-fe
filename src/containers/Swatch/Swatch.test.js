import React from 'react';
import { shallow } from 'enzyme';
import {mapDispatchToProps, Swatch} from './Swatch.js'


describe('Swatch', () => {
  let hexLight;
  let hexDark;

  beforeEach(() => {
    hexLight = 'ffffff'
    hexDark = '000000'
  })
  it('should match the snapshot', () => {
    let wrapper = shallow(<Swatch hex={hexLight} isLocked={true}  num={1}/>)
    expect(wrapper).toMatchSnapshot();
  })
  it('should match the snapshot', () => {
    let wrapper = shallow(<Swatch hex={hexDark} isLocked={false}  num={1}/>)
    expect(wrapper).toMatchSnapshot();
  })



})