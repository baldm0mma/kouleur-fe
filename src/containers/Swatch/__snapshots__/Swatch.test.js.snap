// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swatch should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Swatch should match the snapshot 2`] = `ShallowWrapper {}`;
